app/domain/authentication/usecase/validates_token.go

```diff
@@ -3,21 +3,77 @@ package usecase
 import (
 	"errors"
 	"github.com/golang-jwt/jwt"
+	"github.com/patriciapedrosaa/transfer-me/app/domain/entities"
+	"time"
 )
 
-var ErrInvalidToken = errors.New("unauthorized")
+var (
+	ErrInvalidToken = errors.New("unauthorized")
+	ErrTokenExpired = errors.New("token expired")
+)
 
-func (a Authentication) ValidatesToken(tokenString string) (*jwt.Token, error) {
+func (a Authentication) ValidatesToken(tokenString string) (entities.Token, error) {
 	keyFunc := func(token *jwt.Token) (interface{}, error) {
 		_, ok := token.Method.(*jwt.SigningMethodHMAC)
 		if !ok {
 			return nil, ErrInvalidToken
 		}
 		return []byte(a.accessSecret), nil
 	}
-	token, err := jwt.Parse(tokenString, keyFunc)
+	jwtToken, err := jwt.Parse(tokenString, keyFunc)
 	if err != nil {
-		return nil, err
+		return entities.Token{}, err
+	}
+
+	claims := jwtToken.Claims.(jwt.MapClaims)
+	token := entities.Token{
+		ID:       jwtToken.Claims.(jwt.MapClaims)["id"].(string),
```

Comment:
wouldn't `jwtToken.Claims.(jwt.MapClaims)["id"].(string)` the same as `claims["id"]`?